src/model/race.ts

```diff
@@ -0,0 +1,57 @@
+enum Attributes {
+    STRENGTH,
+    DEXTERITY,
+    INTELLIGENCE,
+    WISDOM,
+    CONSTITUTION,
+    CHARISMA
+};
+
+enum AbilityRefresh {
+    ATWILL,
+    BATTLE,
+    DAILY
+};
+enum AbilityType {
+    STANDARD,
+    MOVE,
+    QUICK,
+    INTERRUPT,
+    FREE
+};
+enum AbilityTrigger {
+    HIT,
+    DAMAGETAKEN,
+    MISS,
+};
+export interface IRace {
+    bonusstat1: Attributes,
+    bonusstat2: Attributes,
+    abilityrefresh: AbilityRefresh,
+    abilitytype: AbilityType,
+    abilitytrigger: AbilityTrigger,
+    abilitytext: string
+
+};
+
+class dwarf implements IRace {
+    bonusstat1: Attributes;
```

Comment:
I'm not sold on this structure.  Does the interface buy us much?  Are classes overkill here?

Reply:
Does a new character work from a Character class with the base attributes, and then add a race that extends their abilities with the attributes listed in the race (and then the same by class)? Making sure I understand the desired flow.

Reply:
I think so.  The character class holds on to some common information, like AC, PD, MD etc.. as well as what race and class type they are.  The character class probably has it's own AC calculation that asks the class for it's answer, then layers in things like magic armor.  I also am presuming the class (barb/fighter) just holds onto ALL available tallents/spells, and the character holds on to the subset they chose.

Reply:
So from the character class you tell it I'm a barb with heavy armor, a light weapon, and a shield. OK Barb class, based on that and the attribute(str/dex/etc...) that I hand you...what is my AC and dmg numbers.  Great, now I know this character has a magic set of plate armor so I'll take that and add +2.  

I think we have to think through how feats work, and magic items...I left them on the character since they aren't necessarily class specific.

Reply:
In my mind, I'm envisioning you mainly have a list of abilities broken apart by how often you can use them. So..

Daily
------
Stufff

Battle
-------
Stuff

Daily
-----
Stuff

And it keeps track of if you have spent them this battle, or this 'rest' period.
The trick there is we have some powers that come from your race, and some from your class, and perhaps some from feats.  It might be a mess to consolidate it all to a large list...you may forget to always pull in the racial power for example

Reply:
And finally. To get to your original question.

Character keeps track of attributes (str,dex,etc..)  it picks a class and race.  The class and race tell it what attributes you can get your +2 bonus on.  The character handles adding them appropriately into the attributes at creation time.